docs/architecture.md

```diff
@@ -0,0 +1,26 @@
+# Architecture
+
+The following diagrams shows preliminary vision for the collaboration between the Filecoin rootnet and the subnets,
+some of which can be implemented using Fendermint:
+
+![Architecture](images/IPC%20with%20Tendermint%20Core.jpg)
+
+The components in a nutshell:
+* __Lotus rootnet__: Lotus has to support IPC to act as the rootnet for its child subnets. There are competing proposals with regards to this being in the form of mostly user-deployed smart contracts, or built-in (privileged) capabilities.
+* __Tendermint Core__: acts as the generic SMR in one of the subnets, talking to all other Tendermint instances in that subnet. It runs separate from the Application, which is completely in our control, and talks to it via ABCI++.
+* __Application__: This is where we implement the IPC ledger logic, the transaction handling, using the FVM/FEVM. We implement the ABCI++ interface to be compatible with Tendermint. Other than that we delegate to reusable smart contracts, for example to produce checkpoints. We rely on ABCI to pass us the headers to be signed, and we can use the ledger to gather signatures for checkpoints.
+* __Parent View__: The Application might observe the parent subnet consensus state directly for the purpose of voting. The goal here is to deal with the fact that Lotus might roll back a message, so we can’t execute a top-down message as soon as it appears, we have to wait for it to be embedded. We can make sure of this either by a) including a light client of the parent in the child ledger or b) using the voting process and let validators take a peek at the other chain.
+* __IPLD Resolver__: A separate process (or a library) that the Application can contact to resolve CIDs and store them in the IPLD Store. It maintains connection with other nodes in the parent/child subnets, or perhaps even beyond.
+* __IPLD Store__: A common store available for read/write for both the FVM and the IPLD Resolver. There might be a separate instance for each subnet, or one larger serving the needs of all subnets of an operator.
+* __FVM__: This is our execution layer. Subnets can use FEVM if they want.
```

Comment:
I think FVM is more a requirement than an option, or at least subnets need to know how IPC messages can be executed against their own specific state (if they use some other kind of runtime ex-FVM).

Reply:
All the above :) 
- FVM is probably a hard requirement for the project but technically an option in future subnets. And you are right, FEVM is an option because its only an actor that we could include in the bundle (I read FVM instead of FEVM :facepalm:)